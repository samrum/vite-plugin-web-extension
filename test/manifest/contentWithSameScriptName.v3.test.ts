import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("contentWithSameScriptName");

runManifestV3Test("contentWithSameScriptName", () => ({
  content_scripts: [
    {
      js: [`${resourceDir}/content1/content.js`],
      matches: [
        "*://*/*",
        "https://*/*",
        "*://example.com/",
        "https://example.com/",
        "*://example.com/subpath/*",
        "https://example.com/subpath/*",
      ],
    },
    {
      js: [`${resourceDir}/content2/content.js`],
      matches: [
        "*://*/*",
        "https://*/*",
        "*://example.com/",
        "https://example.com/",
        "*://example.com/subpath/*",
        "https://example.com/subpath/*",
      ],
    },
  ],
}));
