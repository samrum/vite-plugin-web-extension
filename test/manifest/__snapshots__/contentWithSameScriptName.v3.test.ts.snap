// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contentWithSameScriptName - Manifest V3 1`] = `
[
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  "version": "1.0.0",
  "name": "Manifest Name",
  "manifest_version": 3,
  "content_scripts": [
    {
      "js": [
        "test/manifest/resources/contentWithSameScriptName/content1/content.js"
      ],
      "matches": [
        "*://*/*",
        "https://*/*",
        "*://example.com/",
        "https://example.com/",
        "*://example.com/subpath/*",
        "https://example.com/subpath/*"
      ]
    },
    {
      "js": [
        "test/manifest/resources/contentWithSameScriptName/content2/content.js"
      ],
      "matches": [
        "*://*/*",
        "https://*/*",
        "*://example.com/",
        "https://example.com/",
        "*://example.com/subpath/*",
        "https://example.com/subpath/*"
      ]
    }
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithSameScriptName/content1/content.js",
    "name": undefined,
    "source": "console.log("content1");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithSameScriptName/content2/content.js",
    "name": undefined,
    "source": "console.log("content2");
",
    "type": "asset",
  },
]
`;
